compaction/compaction_manager.cc

```diff
@@ -1962,9 +1962,6 @@ compaction_backlog_tracker::compaction_backlog_tracker(compaction_backlog_tracke
 compaction_backlog_tracker&
 compaction_backlog_tracker::operator=(compaction_backlog_tracker&& x) noexcept {
     if (this != &x) {
-        if (auto manager = std::exchange(_manager, x._manager)) {
-            manager->remove_backlog_tracker(this);
-        }
```

Comment:
I don't know if this is correct. The manager will still refer to `this`, but `this->_manager` will point somewhere else.

We should unregister `this` from the current manager, unregister x from x._manager, adopt x._manager, and register ourselves in it.

Reply:
why? this->manager remains unchanged with this change.
x gets unregistered when destroyed, but I don't think this ever happens since we're assigning registered trackers with unregistered ones, we can even throw internal error if that's not the case.  We may want to move-construct registered trackers, but otherwise the move assignment is to replace the contents of the assigned tracker, not its registration as far as I could tell.

Reply:
In the next version I'll introduce a new function to do this: `transfer_backlog`, and delete the move-assignment operator to avoid confusion.